src/main/java/mate/academy/rickandmorty/service/CharacterService.java

```
@@ -0,0 +1,83 @@
+    private static final String CHARACTER = "character";
+    private static int countOfCharacters = 0;
+    private final ObjectMapper objectMapper;
+    private final CharacterMapper characterMapper;
+    private final CharacterRepository characterRepository;
+
+    public void loadAllCharacters() {
+        if (characterRepository.count() > 0) {
+            return;
+        }
+        HttpClient httpClient = HttpClient.newHttpClient();
+        String url = DEFAULT_API + CHARACTER;
+
+        HttpRequest httpRequest = HttpRequest.newBuilder()
+                .GET()
+                .uri(URI.create(url))
+                .build();
+
+        try {
+            HttpResponse<String> response = httpClient
+                    .send(httpRequest, HttpResponse.BodyHandlers.ofString());
+            ResponseDataDto responseDataDto = objectMapper
+                    .readValue(response.body(), ResponseDataDto.class);
+            List<Character> characterList = new ArrayList<>(responseDataDto.results().stream()
+                    .map(characterMapper::toModel)
+                    .toList());
+
+            while (responseDataDto.info().next() != null) {
+                url = responseDataDto.info().next();
+                httpRequest = HttpRequest.newBuilder().GET().uri(URI.create(url)).build();
+                response = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+                responseDataDto = objectMapper.readValue(response.body(), ResponseDataDto.class);
+                characterList.addAll(responseDataDto.results().stream()
+                        .map(characterMapper::toModel)
+                        .toList());
+            }
+            characterRepository.saveAll(characterList);
+            countOfCharacters = responseDataDto.info().count();
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
```

Exception handling: in the catch block you rethrow a RuntimeException. That will fail application startup if the remote fetch fails. If that is intended (fail fast) it's okay, but consider handling transient network errors (retry, backoff) or at minimum logging a clear error message so test environments and CI understand why startup failed. If you prefer graceful degradation, catch and log and allow the app to start with an empty DB (but adjust controller behavior accordingly).